airgun/browser.py

```diff
@@ -409,3 +412,49 @@ def save_downloaded_file(self, file_uri=None, save_path=None):
         self.url = current_url
         self.plugin.ensure_page_safe()
         return file_path
+
+    def check_alert(self, locator):
+        try:
```

Comment:
Consider moving the modal map with locators into this function, so it handles checking the potential non-browser alert element types internally, returning the relevant widget instance or raising.

Then test authors have the ability to directly validate that the alert is using a non-alert modal by calling this function to get the alert widget.

Reply:
(non-blocking)

Reply:
The reason was designed like that, we basically do not know which type of confirmation modal it will be on the webpage. I do want to continue the loop even after the exception. As per my thinking, `get_alert` will anytime return the right alert object either javascript alert of pf3/4 alert